File: app/src/main/java/dev/bee/kanjianki/MainActivityStudyReviewWriter.java
package dev.bee.kanjianki;

import dev.bee.kanjianki.core.RecordsSchedulerModels;
import dev.bee.kanjianki.core.RecordsStudyModels;

final class MainActivityStudyReviewWriter implements StudyReviewActions.ReviewWriter {
    private final MainActivityStudy activity;

    MainActivityStudyReviewWriter(MainActivityStudy activity) {
        this.activity = activity;
    }

    @Override
    public void saveStudyItem(RecordsStudyModels.StudyItem item) {
        activity.store.saveStudyItem(item);
    }

    @Override
    public void saveReview(
            RecordsSchedulerModels.ReviewRequest request,
            String appliedRating,
            long reviewedAt,
            RecordsStudyModels.StudyItem beforeReview,
            RecordsStudyModels.StudyItem afterReview
    ) {
        activity.store.saveReview(request, appliedRating, reviewedAt, beforeReview, afterReview);
    }
}

File: app/src/main/kotlin/dev/bee/kanjianki/MainActivityStudyReviewWriter.kt
package dev.bee.kanjianki

import dev.bee.kanjianki.core.RecordsSchedulerModels
import dev.bee.kanjianki.core.RecordsStudyModels

internal class MainActivityStudyReviewWriter(
    private val activity: MainActivityStudy,
) : StudyReviewActions.ReviewWriter {
    override fun saveStudyItem(item: RecordsStudyModels.StudyItem) {
        activity.store.saveStudyItem(item)
    }

    override fun saveReview(
        request: RecordsSchedulerModels.ReviewRequest,
        appliedRating: String,
        reviewedAt: Long,
        beforeReview: RecordsStudyModels.StudyItem?,
        afterReview: RecordsStudyModels.StudyItem?,
    ) {
        activity.store.saveReview(request, appliedRating, reviewedAt, beforeReview, afterReview)
    }
}
